selfhost/lexer.jakt

```diff
@@ -332,10 +332,10 @@ enum NumericConstant {
     }
 }
 
-function make_float_token(number: f64, suffix: LiteralSuffix, span: Span) throws -> Token => match suffix {
+function make_float_token(number: f64, suffix: LiteralSuffix, raw_number: String, span: Span) throws -> Token => match suffix {
     // FIXME: consider unifying with make_integer_token() to a single function
-    LiteralSuffix::F32 => Token::Number(number: NumericConstant::F32(f64_to_f32(number)), span)
-    LiteralSuffix::F64 => Token::Number(number: NumericConstant::F64(number), span)
+    LiteralSuffix::F32 => Token::Number(number: NumericConstant::F32(f64_to_f32(number)), raw_number, span)
+    LiteralSuffix::F64 => Token::Number(number: NumericConstant::F64(number), raw_number, span)
```

Comment:
this makes me wonder if we should just keep Token::Number a string and let the parser handle actually parsing it.

Reply:
Yeah it probably does make more sense, was thinking that while making the changes.